### ISSUE TITLE
Checks > References > validation don't recognize locale references

### ISSUE THREAD

**Issue by LAfricain** [postiffm/bibledit-desktop#61]
For French (maybe for French Catholics) we use the ref like that: Mt 1,3-18.20, that means  chap 1 vv. 3 till  18 and only v. 20. If I have ref with like that : `(Lk 5,26 ; 7,16 ; 8,25.35.37)` the check considers it as an error: Lk 35.37 is read like Luc 25:35 but it is the verses 35 and 37.
"," is chap and verse separator,  "-" is for verset range, and "." is verse separator.

**Comment by postiffm** [comment]
So English Mt 1:3-18,20 = French Mt 1,3-18.20. But where did the 25 in Luc 25 come from?

**Comment by LAfricain** [comment]
> So English Mt 1:3-18,20 = French Mt 1,3-18.20.

OK
> But in Lk 35.37, where does the 25 come from?

It's a mistake, it read 25:35
> And what does ";" mean in French?

That means "new reference". In our case, Lk chap 7 v. 16 and Lk chap 8 v.25 v.35 v.37

**Comment by LAfricain** [comment]
For italian, kikongo, lingala, and maybe other Catholic bible the "," is also used between chap and verses. But for English Catholic bible, it is the ":".

**Comment by rluzynski** [comment]
Is this a language-specific difference or a denomination-specific, or maybe both? If language-specific then the format differences should be handled by internationalization. If denomination-specific then each Bible text should be treated differently. If both then the denomination-specific markers for each Bible text should be further passed through i18n subsystem.

**Comment by LAfricain** [comment]
I'm not sure... Because in English, it is for all denomination the same. But in French I think (but not sure) Protestant use : and catholic , I will ask friends and come back.

**Comment by LAfricain** [comment]
According this [page](https://exegeseettheologie.wordpress.com/2011/02/15/elements-de-methodologie-pour-la-redaction-de-travaux-de-niveau-universitaire/) (in French), it seems to be more a language-specific difference. Gn 1:1-3 for english language contries, Gn 1,1-3 for French speaking contries. A third possibility not specific to a language: Gn 1.1-3. But I'm not sure it is good to "code" it according the localization, because if you want to use bibledit-desktop for an exotic language (it is his goal), how to do it if it is not localized? If you choose the localization way, you need to add all the local languages of the French African contries... Italian is the same way as French.

**Comment by rluzynski** [comment]
I've checked a [scientific source](http://sjp.pwn.pl/poradnia/haslo/Jak-cytowac-Pismo-Swiete;14204.html) for my language and they say that there are multiple conventions and all are acceptable. They only emphasize that one should consequently use one convention across whole single publication. Let's assume that Bibledit Desktop is a single publication but different translations are different publications (like when you have a translated book you normally expect the translator to follow the conventions of your language rather than the conventions of the original language). So I think it's fine to apply one convention to whole Bibledit Desktop but to allow different conventions for different languages.

> A third possibility not specific to a language: Gn 1.1-3.

I think there are no "not specific to a language" conventions, at least I don't think we should provide any here. An application always uses a language, default is usually English so I think it is reasonable to use English conventions for English locales.

> But I'm not sure it is good to "code" it according the localization, because if you want to use bibledit-desktop for an exotic language (it is his goal), how to do it if it is not localized? If you choose the localization way, you need to add all the local languages of the French African contries... Italian is the same way as French.

In case of other languages you have two answers for potential users, both are correct:

1. Your language is not supported, the application runs in English and you must accept that the references are English as well.
2. This is open source, feel free to provide your own translation. (At least the translation of the reference format.)

**Comment by LAfricain** [comment]
I'm a little lost in all your reasoning, I just have a problem:
  It is traditional in French to use a reference type, and translating bibledit has no impact on that. And so if this type of reference is not taken into account, the option of checking references is absolutely useless. Why penalize potential users of bibledit?

> Let's assume that Bibledit Desktop is a single publication but different translations are different publications

I don't agree. Bibleidit can be used by differents societies. You can not force all societies to adopt a unique way of doing this. For example, I use also bibledit with already published bibles, for the creation of sword modules. The reference are already wrote.

> Your language is not supported, the application runs in English and you must accept that the references are English as well.

???

> This is open source, feel free to provide your own translation. (At least the translation of the reference format.)

It is already translated in French, it has nothing to do with our problem. The check works on the text translated in the bibledit, currently I use on it the Kikongo, and we write references like in French. and I'm not a dev. ;)

**Comment by LAfricain** [comment]
Yes :) Joy :) Now you understood well!! Thanks for patience. I agree now all your propositions about conventions.
I think there are 3 ways to mark ref, but I discovered one new yesterday... → ex Mt 1 12-15 (Mt chap 1 verses 12 to 15, in French BJ), but it's not very common.
